project/src/main/java/com/google/autograder/servlets/ListAssignmentSubmissionsServlet.java

```diff
@@ -2,23 +2,82 @@
 
 import java.net.URL;
 import java.io.IOException;
+import java.net.HttpURLConnection;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.annotation.WebServlet;
 import com.google.autograder.data.Database;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import com.google.appengine.api.datastore.Query;
 import com.google.appengine.api.datastore.Entity;
+import com.google.autograder.servlets.helpers.API;
 import com.google.appengine.api.datastore.Query.Filter;
 import com.google.appengine.api.datastore.PreparedQuery;
 import com.google.appengine.api.datastore.Query.FilterOperator;
 import com.google.appengine.api.datastore.Query.FilterPredicate;
 
-@WebServlet("/listAssignmentSubmissions")
+// @WebServlet("/listAssignmentSubmissions")
 public final class ListAssignmentSubmissionsServlet extends HttpServlet {
 
+    private static String END_POINT = "https://classroom.googleapis.com/v1/courses/{courseId}/courseWork/{courseWorkId}/studentSubmissions";
+
     @Override
     protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        if (Database.getUserService().isUserLoggedIn()) {
```

Comment:
Prefer to exit early rather than to keep nesting:
```
if (!Database.isUserLoggedn()) {
  response.setHeader("redirect", ...);
}

if (authorization == null) {
  repsonse.setHeader("redirect", ...);
}

// rest of the code
```

Reply:
Ok no problem, that makes sense. It's a lot cleaner too.